docs/source/jit_python_values.rst

```diff
@@ -0,0 +1,466 @@
+.. contents::
+    :local:
+    :depth: 2
+
+
+.. testsetup::
+
+    # These are hidden from the docs, but these are necessary for `doctest`
+    # since the `inspect` module doesn't play nicely with the execution
+    # environment for `doctest`
+    import torch
+
+    original_script = torch.jit.script
+    def script_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_script(obj, *args, **kwargs)
+    torch.jit.script = script_wrapper
+
+    original_trace = torch.jit.trace
+    def trace_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_trace(obj, *args, **kwargs)
+    torch.jit.trace = trace_wrapper
+
+.. _python-builtin-functions-values-resolution:
+
+Resolution Rules
+================
+When given a Python value, TorchScript attempts to resolve them in following five different ways:
+
+* Compilable Python Implementation: 
+    * When a Python value is backed by a Python implementation that can be compiled by TorchScript, TorchScript compiles and uses the underlying Python function.
+    * Example: ``torch.jit.Attribute``
+* Op Python Wrapper: 
+    * When a Python value is a wrapper of a native PyTorch op, TorchScript emits corresponding operator.
+    * Example: ``torch.jit._logging.add_stat_value``
+* Python Object Identity Match: 
+    * For a limited set of ``torch.*`` API calls (in the form of Python values) that TorchScript supports, TorchScript attempts to match a Python value against each item in the set. 
+    * When matched, TorchScript generates a corresponding SugaredValue instance that contains lowering logic for these values.
```

Comment:
There's no explanation in other parts of the spec what `SugaredValue` is, so this might be confusing.
```suggestion
    * When matched, TorchScript generates a corresponding ``SugaredValue`` instance that contains lowering logic for these values.
```

Reply:
Added code formatting per suggested fix.

Are there any existing doc on what SugaredValue is? if not, I could add a link pointing to it, otherwise, I could add a few sentences here describing very roughly on what a SugaredValue is.